2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors/scalastream/CollectorService.scala

```diff
@@ -81,22 +84,97 @@ class CollectorService(
           optionalHeaderValueByName("User-Agent") { userAgent =>
             optionalHeaderValueByName("Referer") { refererURI =>
               headerValueByName("Raw-Request-URI") { rawRequest =>
-                hostName { host =>
-                  clientIP { ip =>
-                    requestInstance{ request =>
-                      entity(as[String]) { body =>
+                headerValueByName("Kinesis-Part-Key") { kinesisPartKey =>
```

Comment:
Only difference between optionalHeaderValueByName and headerValueByName is how the resulting value (or lack thereof) is cast